crypto/x509/pcy_node.c

```diff
@@ -93,27 +93,28 @@ X509_POLICY_NODE *ossl_policy_level_add_node(X509_POLICY_LEVEL *level,
         }
     }
 
+    tree->node_count++;
+    if (parent)
+        parent->nchild++;
+
     if (extra_data) {
         if (tree->extra_data == NULL)
             tree->extra_data = sk_X509_POLICY_DATA_new_null();
         if (tree->extra_data == NULL) {
             ERR_raise(ERR_LIB_X509V3, ERR_R_CRYPTO_LIB);
-            goto node_error;
+            goto extra_data_error;
         }
         if (!sk_X509_POLICY_DATA_push(tree->extra_data, data)) {
             ERR_raise(ERR_LIB_X509V3, ERR_R_CRYPTO_LIB);
-            goto node_error;
+            goto extra_data_error;
         }
     }
 
-    tree->node_count++;
-    if (parent)
-        parent->nchild++;
-
     return node;
 
  node_error:
     ossl_policy_node_free(node);
+ extra_data_error:
```

Comment:
It's probably easier just to set `node` to NULL and rely on `ossl_plicy_node_free()` handling NULLs correctly, rather than add this extra label.